ic-os/docs/Network-Configuration.adoc

```diff
@@ -118,4 +118,80 @@ In other words, the prefix of the EUI-64 formatted IPv6 SLAAC address is swapped
+The network part (i.e. +ipv6_prefix+) of the IPv6 address is retrieved from the +config.json+ configuration file. The host part is the EUI-64 formatted address.
+
+== Active backup
+
+[NOTE]
+This feature is currently under development. See ticket https://dfinity.atlassian.net/browse/NODE-869#[NODE-869].
+
+In order to simplify the physical cabling of the machine, Linux's active-backup bonding technique is utilized. This operating mode also improves redundancy when more than one 10-gigabit ethernet network interface is connected to the switch. A Node Technician can decide to either just use one or all of the 10GbE network interfaces in the bond. The handling of the uplink and connectivity is taken care of by the Linux operating system.
+
+Details can be found in:
+
+  ic/rs/ic_os/os_tools/setupos_tool generate-network-config
+
+Note that this mode does not increase the bandwidth/throughput. Only one link will be active at the same time.
+
+== Firewalling
+
+IC-OS supports firewall traffic control from certain hosts or subnets, through
++firewall.json+ in the SetupOS config partition.  This file is copied during setup
+into both the HostOS and the GuestOS config partitions.  The firewall rules
+are specified as a list:
+
+  [
+    {...firewall rule...},
+    {...firewall rule...},
+    ...
+  ]
+
+Each element on the list is a dictionary that contains the following keys:
+
+* `from`: IPv4 / IPv6 address or subnet, with netmask (prefix length) after a slash.
+  Example: `2001:db8:abcd:0012::0/64`.  This is mandatory and it must validate as
+  a valid address and netmask, otherwise firewall configuration will be ineffective.
+* `to`: Either `HostOS` or `GuestOS` or `both` (capitalized that way), or omit it to
+  indicate that the rule applies to both compartments (same as `both`).
+* `protocol`: `tcp`, `udp`, or `all` (default `all`).
+* `from_ports` / `to_ports`: an integer port or a dash-separated port range from zero
+  to 65535, indicating the source or destination ports to be opened or blocked.  This
+  requires that protocol be `tcp` or `udp`, else it is ignored.  If unspecified, all
+  traffic using that protocol is covered by the rule.
+* `action`: Either `accept` or `drop`, or omit it to mean `accept`.  Traffic is
+  normally dropped in almost all circumstances.
+* `comment`: Freeform text that isn't used for anything other than your own reference.
+
+Inbound traffic is processed by the firewall engine in the order of the rules as they
+are seen, but note that rules for IPv6 and IPV4 are processed *separately in order*
+because the IPv6 and IPv4 stacks process packets separately from each other.
+Some rules that IC-OS needs to operate normally will always be processed prior to these
+custom rules.  Any other inbound traffic not covered by the rules will be dropped.
+
+Here is a more complete example of a valid +firewall.json+ snippet:
```

Comment:
@andrewbattat would it be possible to add a template `firewall.json` in the config partition, to make it easier for the NP to create their own configuration?

Reply:
We've added an empty one, but you can't add a template anything with JSON because, well, JSON cannot have anything templated or commented.

Reply:
@DFINITYManu So long as 

1. The wiki documentation is clear for how to write the firewall.json files
2. SetupOS fails early if the firewall.json format is invalid

Then I think we're good. 

It's difficult creating a template because the different formats of the firewall.json file is infinite (How many rules do we template? Do we template for every key, like protocol?)

Reply:
Yeah a template does not make much sense.  And the wiki should gain the documentation once whatever part of the merge train updates it.

Reply:
> And the wiki should gain the documentation once whatever part of the merge train updates it.
We manually maintain the wiki. Both node deployment guides will have to be updated:

- https://wiki.internetcomputer.org/wiki/Node_Deployment_Guide#6._Add_configuration
- https://wiki.internetcomputer.org/wiki/Node_Deployment_Guide_(with_an_HSM)#6._Add_configuration

And the networking guide:

- https://wiki.internetcomputer.org/wiki/Node_Provider_Networking_Guide

Reply:
Will do.  Tracked in https://dfinity.atlassian.net/browse/DRE-258 .